### ISSUE TITLE
Tests can't be run concurrently from different builds

### ISSUE THREAD

**Issue by riastradh** [heimdal/heimdal#1172]
**Describe the bug**
The Heimdal tests bind to fixed port numbers during the test suite:

https://github.com/heimdal/heimdal/blob/ca0e53b64fa88e6d3fe5a04292bd7ec79b8081de/tests/kdc/Makefile.am#L49-L56
https://github.com/heimdal/heimdal/blob/ca0e53b64fa88e6d3fe5a04292bd7ec79b8081de/tests/kdc/check-cc.in#L51-L57

Thus, if two instances of the test suite are run concurrently, they may conflict and tests may spuriously fail.

**To Reproduce**
1. Build Heimdal in two trees at once.
2. Run `gmake check` in both trees simultaneously.

**Expected behavior**
Concurrent runs of the test suite pass.

Note: This is an separate issue from a single instance of the test suite failing with parallel make jobs, e.g. `gmake -j4 check`.

**Comment by nicowilliams** [comment]
Yup.  This is a problem.  We really need to rationalize all the tests and have them automatically pick port numbers to use.
